contents/node/nodejs.md

```diff
@@ -0,0 +1,206 @@
+
+## Benefit 9: Support for better code organization
+
+We can also create separate files and modules in our codebase such that each module is focused on a single functionality. This makes your code more maintainable and testable.
+
+To export the `Parser` constructor in `parser.js`,
+```js
+function Parser(options) {
+    this._options = options || {};
+}
+
+Parser.prototype.parse = function (content) {
+  // insert code here
+}
+module.exports = Parser; // override the exports object
+```
+
+The Parser constructor will then be returned whenever the `require` function is used to include the `Parser` module. It can be used to create Parser objects in other files in the project, for example,
+```js
+const parser = require('../parser') // import content from parser.js based on relative file path. The js extension is assumed and can be excluded.
+
+const content = 'some content';
+const newParser = new Parser();
+newParser.parse(content);
+```
+
+To better understand how to use `module.exports`, checkout [Tendai Mutunhire's article](http://stackabuse.com/how-to-use-module-exports-in-node-js/).
+
+# Use cases
+
+Node.js is **good** for
+* **Processing high volumes of IO-bound requests**. A single instance of a Node server will be more efficient and can serve more requests with the same hardware than other usual servers (such as those written in Ruby on Rails). This makes the Node server faster and more scalable.
+* **Real time apps** where you have to process a large volume of requests with little delay. This includes instant messaging apps and collaborative editing apps where you can watch the document being modified live such as Trello and Google Docs. Node.js is a good choice as it can handle multiple client requests (even while waiting for responses).
+* **Single-page apps** where a lot of processing and rendering is done on the client's side and the backend server only needs to provide a simple API. Node can process many requests with low response times. In addition, you can reuse and share code, such as those for validation, between the client and server.
+
+However, Node.js is **not suitable** for
+* **CPU-intensive jobs**. Remember that Node.js is single-threaded. If the thread is busy doing CPU-heavy operations, it will not be able to process incoming requests timely. This removes the biggest advantage of Node. Perhaps, you should consider an alternative technology such as Go which supports concurrency.
+
+
+To find out more about when you should or should not use Node.js, checkout these articles by [netguru](https://www.netguru.co/blog/use-node-js-backend) and [Node.js foundation](https://medium.com/the-node-js-collection/why-the-hell-would-you-use-node-js-4b053b94ab8e).
+
+# Resources
```

Comment:
Is there a way a reader can tryout node without installing it? like an online platform? Just curious.

Reply:
I just went to check and there is [RunKit]. It should work in the browser and it has all the modules/versions pre-installed.

Prof @damithc, the rest of the content has been updated.